libs/community/langchain_community/cache.py

```diff
@@ -1034,36 +1034,71 @@ class CassandraCache(BaseCache):
     """
     Cache that uses Cassandra / Astra DB as a backend.
 
+    Example:
+
+        .. code-block:: python
+
+            import cassio
+
+            from langchain_community.cache import CassandraCache
+            from langchain_core.globals import set_llm_cache
+
+            cassio.init(auto=True)  # Requires env. variables, see CassIO docs
+
+            set_llm_cache(CassandraCache())
+
     It uses a single Cassandra table.
     The lookup keys (which get to form the primary key) are:
         - prompt, a string
         - llm_string, a deterministic str representation of the model parameters.
-          (needed to prevent collisions same-prompt-different-model collisions)
+          (needed to prevent same-prompt-different-model collisions)
+
+    Args:
+        table_name: name of the Cassandra table to use as cache
+        session: an open Cassandra session.
+            Leave unspecified to use the global cassio init (see below)
+        keyspace: the keyspace to use for storing the cache.
+            Leave unspecified to use the global cassio init (see below)
+        ttl_seconds: time-to-live for cache entries
+            (default: None, i.e. forever)
+        skip_provisioning: you can pass True if the Cassandra
+            table is guaranteed to exist already for a faster initialization.
+
+    Note:
+        The session and keyspace parameters, when left out (or passed as None),
+        fall back to the globally-available cassio settings if are available.
+        In other words, if a previously-run 'cassio.init(...)' has been
+        executed previously anywhere in the code, Cassandra-based objects
+        need not specify the connection parameters at all.
     """
 
     def __init__(
         self,
+        table_name: str = CASSANDRA_CACHE_DEFAULT_TABLE_NAME,
```

Comment:
this is a breaking change, would prefer to avoid it

Reply:
This is, in a sense, the point of this PR. Using the `cassio.init()` construct to set a globally-available database connection would enable a much easier constructor pattern where one can omit the session/keyspace arguments altogether when creating table-based LangChain objects.

I am aware this is a breaking change, on the other hand it would make it possible to use this pattern like `CassandraCache("table")` instead of the more unwieldy `CassandraCache(None, None, "table")`, something that would really be an advantage in the long run.

Would you have a suggestion on how to best make this transition happen by respecting legacy usage as much as possible?